### DISCUSSION TITLE
AWS ECS & ALB & CloudFront. sometimes happed status code 502

### DISCUSSION THREAD

**Discussion by eretica** [vercel/next.js#16544] [Help]
I'm using Nextjs in a configuration called ECS ALB CloudFront using AWS

My production environment may return 502 several times a day when returning NextAPIRoute or a file

I checked the situation in Athena and it looks like a screenshot.

<img width="1873" alt="スクリーンショット 2020-08-25 14 36 00" src="https://user-images.githubusercontent.com/16332352/91127353-07233780-e6e1-11ea-9140-4546e080a0ba.png">

I'm guessing that keepalive is the cause, but is there a solution?

https://shuheikagawa.com/blog/2019/04/25/keep-alive-timeout/
Like this article.

```
server.keepAliveTimeout = 61 * 1000;
server.headersTimeout = 65 * 1000; // This should be bigger than `keepAliveTimeout + your server's expected response time`
```

We think we can improve this by setting up

Or, if you have any other suggestions, please let me know.

**Comment by jordie23** [comment]
Heya the only solution I was able to find was to modify the IdleTimeout on the AWS ELB to be 4 seconds, i.e 1 second less than node's default 5 seconds.

**Comment by tabuchid** [comment]
I tried making a feature request but it got converted to a discussion https://github.com/vercel/next.js/discussions/28642
Maybe we'll get some answers there.

**Comment by schorfES** [comment]
This PR (https://github.com/vercel/next.js/pull/35827) introduces a change to the CLI that allows to set `keepAliveTimeout`:

```bash
next start --keepAliveTimeout 65000
```
